### ISSUE TITLE
Feedbacks?

### ISSUE THREAD

**Issue by nevillelyh** [nevillelyh/scio-koans#2]
Please give feedback on the entire collection of Koans overall:
- Structure
- Continuity
- User experience
- Missing topics
- What else?

And on individual Koans:
- Too easy/hard?
- Too little/much hint?
- Context?
- What else?

**Comment by RickardZwahlen** [comment]
There's a pesky warning that is pretty spammy:
```[info] /home/zwahlen/Projects/scio-koans/src/test/scala/scio/koans/a2_coders/K00_Avro.scala:60:48: 
[info] Using a fallback coder for Avro's GenericRecord is discouraged as it is VERY inefficient.
[info] It is highly recommended to define a proper Coder[GenericRecord] using:
[info] 
[info]   Coder.avroGenericRecordCoder(schema)
[info]         
[info]   val scioGeneric: Coder[GenericRecord] = Coder[GenericRecord]```

It would also be great if the loop could print which Koan it is currently at, or something to track progress.

**Comment by nevillelyh** [comment]
Added some progress info:
cee112cbfc5437241c83efc714990626dac41f80
37243d961d8c440e77eaa415fe3f6af54e7225c0

The coder warning is tricky. It seems to be caused by `Jmh / compile` which rebuilds everything every time. I'll look into calling that only when the current test is a `JmhKoan`.

**Comment by nevillelyh** [comment]
The compilation issue should be fixed by 5a457a3d81d53ce824a7120226aa2de7fa84eba4.

**Comment by pacuna** [comment]
Hi @nevillelyh, some of the examples are not passing the benchmark tests. One example is the second test for `K02_Converter2`. My solution was failing"

```
K02_Converter2:
[info] Revisions
[info] - should return the same result as baseline
[info] - should be faster than baseline *** FAILED ***
[info]   Benchmark v2: 6353.159808376573 was not greater than or equal to 18213.053241034377 (JmhKoan.scala:103)
```

And I checked the diff and it was the same answer as the solution diff. I had the same issue with other exercises as well.
I can document the other ones I've found if you want to.

Edit: Might be the versions?:

```
sbt version in this project: 1.6.2
openjdk version "11.0.14" 2022-01-18 LTS
Scala code runner version 3.1.1 -- Copyright 2002-2022, LAMP/EPFL
```

**Comment by nevillelyh** [comment]
@pacuna yeah these JMH benchmarks are not deterministic and result can vary due to JVM, system load, etc.
You can increase their chance of passing by increasing bench iterations or time but that also slows down iteration speed.
https://github.com/nevillelyh/scio-koans/blob/main/src/test/scala/scio/koans/shared/JmhKoan.scala#L29-L31
